spectrocloud/resource_cluster_vsphere_test.go

```diff
@@ -0,0 +1,741 @@
+	masterInstance = append(masterInstance, map[string]interface{}{
+		"disk_size_gb": 40,
+		"memory_mb":    8192,
+		"cpu":          4,
+	})
+	mPools = append(mPools, map[string]interface{}{
+		"control_plane":           true,
+		"control_plane_as_worker": true,
+		"name":                    "master-pool",
+		"count":                   1,
+		"placement":               masterPlacement,
+		"instance_type":           masterInstance,
+	})
+
+	// Adding Worker pool
+	workerPlacement := make([]interface{}, 0)
+	workerPlacement = append(workerPlacement, map[string]interface{}{
+		"id":                "",
+		"cluster":           "test cluster",
+		"resource_pool":     "Default",
+		"datastore":         "datastore55_2",
+		"network":           "VM Network",
+		"static_ip_pool_id": "testpoolid",
+	})
+
+	workerInstance := make([]interface{}, 0)
+	workerInstance = append(workerInstance, map[string]interface{}{
+		"disk_size_gb": 40,
+		"memory_mb":    8192,
+		"cpu":          4,
+	})
+
+	mPools = append(mPools, map[string]interface{}{
+		"control_plane":           false,
+		"control_plane_as_worker": false,
+		"name":                    "worker-basic",
+		"count":                   1,
+		"placement":               workerPlacement,
+		"instance_type":           workerInstance,
+	})
+	d.Set("machine_pool", mPools)
+	return d
+}
+
+func TestToVsphereCluster(t *testing.T) {
+	assert := assert.New(t)
+	// Create a mock ResourceData object
+	d := prepareClusterVsphereTestData()
+
+	m := &client.V1Client{
+		GetClusterClientFn: func() (clusterC.ClientService, error) {
+			return &clusterC.Client{}, nil
+		},
+		GetClusterWithoutStatusFn: func(uid string) (*models.V1SpectroCluster, error) {
+			return &models.V1SpectroCluster{
+				Metadata: nil,
+				Spec:     nil,
+			}, nil
+		},
+	}
+
+	vSphereSchema := toVsphereCluster(m, d)
+
+	// Check the output against the expected values
+	// Verifying cluster name attribute
+	assert.Equal(d.Get("name"), vSphereSchema.Metadata.Name)
+
+	// Verifying cluster name attribute
+	assert.Equal("vmware-basic-infra-profile-id", vSphereSchema.Spec.Profiles[0].UID)
+
+	// Verifying account id attribute
+	assert.Equal("vmware-basic-account-id", vSphereSchema.Spec.CloudAccountUID)
+
+	// Verifying cloud config attributes
+	assert.Equal("spectrocloud.dev", vSphereSchema.Spec.CloudConfig.ControlPlaneEndpoint.DdnsSearchDomain)
+	assert.Equal("DDNS", vSphereSchema.Spec.CloudConfig.ControlPlaneEndpoint.Type)
+	assert.Equal("Datacenter", vSphereSchema.Spec.CloudConfig.Placement.Datacenter)
+	assert.Equal("sc_test/terraform", vSphereSchema.Spec.CloudConfig.Placement.Folder)
+	assert.Equal("ssh-rsa AAAAB3NzaC1y", vSphereSchema.Spec.CloudConfig.SSHKeys[0])
+	assert.Equal(false, vSphereSchema.Spec.CloudConfig.StaticIP)
+
+	// Verifying Master pool attributes
+	assert.Equal("master-pool", *vSphereSchema.Spec.Machinepoolconfig[0].PoolConfig.Name)
+	assert.Equal(true, vSphereSchema.Spec.Machinepoolconfig[0].PoolConfig.IsControlPlane)
```

Comment:
Is it possible that pools will change order? May be better to rewrite assert no matter on what is the order.

Reply:
Handled change order cases for master and worker pool